src/ui_basic/editbox.cc

```diff
@@ -212,6 +213,55 @@ bool EditBox::handle_mousepress(const uint8_t btn, int32_t, int32_t) {
 	return false;
 }
 
+void EditBox::set_caret_to_cursor_pos(int32_t x) {
+	if (x <= 2 * kMarginX) {
+		set_caret_pos(0);
+		return;
+	}
+
+	int text_w = text_width(m_->text, *m_->font_style, m_->font_scale);
+	if (text_w <= 0) {
```

Comment:
// no visible text

Reply:
improved by merging the two boundary checks into one. should read good enough without any additional comments